woocommerce/woocommerce
# Prevent PHP `doing_it_wrong` warnings in extend-cart-checkout-block package

### Changes proposed in this Pull Request:

Updates the `extend-cart-checkout` package to register the `checkout-newsletter-subscription-block` during `init` rather than `woocommerce_blocks_loaded`. The latter caused a warning as it was registering scripts too early.

I also updated the version of prettier-config because without being pinned to that version, dependency resolution errors occurred during install.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to `wp-content/plugins` and run `npx @wordpress/create-block -t woocommerce-monorepo/packages/js/extend-cart-checkout-block/ example-checkout-block` (update the path as appropriate!)
2. Go to WP Dashboard -> Plugins and activate the `Example Checkout Block` plugin.
3. Ensure no PHP or JS warnings/errors appear during the following steps
4. Go to the site editor and view the Cart page (verify no errors or warnings appear)
5. Go to the Checkout page. Verify this checkbox appears:
<img width="633" alt="image" src="https://github.com/user-attachments/assets/0c393af2-948f-46e9-bcb3-8b06c969dd8c">

6. Go to the Cart page on the front end, ensure no errors or warnings appear.
7. Go to the Checkout page, ensure the checkbox is there, ensure you can toggle it, and ensure that you cannot place the order while it is unticked. Tick the box and ensure the order is placed.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
